src/server/helpers/middleware.ts

```diff
@@ -131,67 +130,67 @@ export async function loadWorkTableAuthors(req: $Request, res: $Response, next:
+export async function addRelationships(entity, relationshipSet, orm) {
 	async function getEntityWithAlias(relEntity) {
 		const redirectBbid = await orm.func.entity.recursivelyGetRedirectBBID(orm, relEntity.bbid, null);
 		const model = commonUtils.getEntityModelByType(orm, relEntity.type);
 
 		return model.forge({bbid: redirectBbid})
 			.fetch({require: false, withRelated: ['defaultAlias'].concat(utils.getAdditionalRelations(relEntity.type))});
 	}
+
 	entity.relationships = relationshipSet ?
 		relationshipSet.related('relationships').toJSON() : [];
 
 	utils.attachAttributes(entity.relationships);
 
-
 	/**
 	 * Source and target are generic Entity objects, so until we have
 	 * a good way of polymorphically fetching the right specific entity,
 	 * we need to fetch default alias in a somewhat sketchier way.
 	 */
-	return Promise.all(entity.relationships.map((relationship) =>
-		Promise.all([getEntityWithAlias(relationship.source), getEntityWithAlias(relationship.target)])
-			.then(([source, target]) => {
-				relationship.source = source.toJSON();
-				relationship.target = target.toJSON();
-
-				return relationship;
-			})));
+	const relationshipPromises = entity.relationships.map(async (relationship) => {
+		const [source, target] = await Promise.all([getEntityWithAlias(relationship.source), getEntityWithAlias(relationship.target)]);
+		relationship.source = source.toJSON();
+		relationship.target = target.toJSON();
+
+		return relationship;
+	});
+
+	return Promise.all(relationshipPromises);
 }
 
-export function loadEntityRelationships(req: $Request, res: $Response, next: NextFunction) {
+
+export async function loadEntityRelationships(req: $Request, res: $Response, next: NextFunction) {
 	const {orm}: any = req.app.locals;
 	const {RelationshipSet} = orm;
 	const {entity} = res.locals;
 
-	new Promise<void>((resolve) => {
+	try {
 		if (!entity) {
 			throw new error.SiteError('Failed to load entity');
 		}
 
-		resolve();
-	})
-		.then(
-			() => RelationshipSet.forge({id: entity.relationshipSetId})
-				.fetch({
-					require: false,
-					withRelated: [
-						'relationships.source',
-						'relationships.target',
-						'relationships.type.attributeTypes',
-						'relationships.attributeSet.relationshipAttributes.value',
-						'relationships.attributeSet.relationshipAttributes.type'
-					]
-				})
-		)
-		.then((relationshipSet) => addRelationships(entity, relationshipSet, orm))
-		.then(() => {
-			next();
-			return null;
-		})
-		.catch(next);
+		const relationshipSet = await RelationshipSet.forge({id: entity.relationshipSetId})
+			.fetch({
+				require: false,
+				withRelated: [
+					'relationships.source',
+					'relationships.target',
+					'relationships.type.attributeTypes',
+					'relationships.attributeSet.relationshipAttributes.value',
+					'relationships.attributeSet.relationshipAttributes.type'
+				]
+			});
+
+		await addRelationships(entity, relationshipSet, orm);
+
```

Comment:
I'm getting another ESLint error ([callback-return](https://github.com/mysticatea/eslint-plugin-node/blob/v11.1.0/docs/rules/callback-return.md)), which we can solve easily by moving this `next();` a coupl eof lines, right after the end of the catch block.

The same applies to the `loaderFunc` function on line 36